src/include/duckdb/execution/operator/projection/OrdinalityData.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "duckdb/common/types/data_chunk.hpp"
+
+namespace duckdb {
+struct OrdinalityData {
+	static void PrepareOrdinality(DataChunk &chunk, const vector<column_t> &column_ids, idx_t &ord_index) {
```

Comment:
I'm not a huge fan of making this a static method, instead I think it's better if OrdinalityData has `ord_index` and `ord_reset`, and this method renamed to SetOrdinality.
I also think the `ordinality_column` should be set in the OrdinalityData struct, I can imagine this breaking if multiple modifiers add a chunk and they both expect to be the last column in the chunk.

On that note, can we add an assertion that the `chunk.data[ordinality_column]` has the type we expect it to have?

Reply:
I don't think they need to be separate methods then either, `ord_reset` will just stay `false` for the cases where it is not applicable.

Reply:
> On that note, can we add an assertion that the chunk.data[ordinality_column] has the type we expect it to have?

Are you talking about the _LogicalType_ or _vector_type_? 
For the _LogicalType_ I chose **Integer** in _bind_table_function.cpp_, while the _vector_type_ created by duckdb is a **FLAT_VECTOR**. 
Interesting to note here is that the Sequence function turns the **FLAT_VECTOR** into a **SEQUENCE_VECTOR**, but I assume you want the assertion right before executing SetOrdinality.

Do you think checking if the vector is of _LogicalTypeId_::**INTEGER** is enough here, should another already existing _LogicalType_ be used instead of **INTEGER**, or should I maybe even add a new _LogicalType_ for this assertion?

Reply:
Just checking that it's an integer is fine, just in case the creation of the sequence vector doesn't assert this